app/controllers/course/lesson_plan/items_controller.rb

```diff
@@ -42,9 +42,22 @@ def render_json_response
     @folder_loader = Course::Material::PreloadService.new(current_course)
 
     assessment_tabs_titles_hash
+    assessment_tabs_visibility_hash
     render 'index'
   end
 
+  # Returns a hash that maps the array in `assessment_tabs_titles_hash` to its
+  # visiblity setting.
+  # Both the lesson_plan_item_settings and the assessment_tabs_titles_hash contain 1 entry
+  # for each assessment tab in the course.
+  #
+  # @return [Hash{Array<String> => Boolean]
+  def assessment_tabs_visibility_hash
+    @assessment_tabs_visibility_hash = @item_settings.lesson_plan_item_settings.map do |setting|
+      [assessment_tabs_titles_hash[setting[:options][:tab_id]].join(': '), setting[:visible]]
```

Comment:
The backend shouldn't know to `join(': ')`. This is a frontend thing. 

Actually, this implementation detail should be black-boxed in `client/app/bundles/course/lesson-plan/reducers/utils.js`'s `generateTypeKey`. Anyone who wants to generate the key should use that function.